packages/editor-ui/src/views/MfaView.vue

```diff
@@ -0,0 +1,291 @@
+							@click="onRecoveryCodeClick"
+							:class="$style.recoveryCodeLink"
+						>
+							{{ $locale.baseText('mfa.recovery.input.info.action') }}</a
+						>
+					</n8n-text>
+				</div>
+			</div>
+			<div>
+				<n8n-button
+					float="right"
+					:loading="verifyingMfaToken"
+					:label="
+						showRecoveryCodeForm
+							? $locale.baseText('mfa.recovery.button.verify')
+							: $locale.baseText('mfa.code.button.continue')
+					"
+					size="large"
+					:disabled="!hasAnyChanges"
+					@click="onSaveClick"
+				/>
+				<n8n-button
+					float="left"
+					:label="$locale.baseText('mfa.button.back')"
+					size="large"
+					type="tertiary"
+					@click="onBackClick"
+				/>
+			</div>
+		</n8n-card>
+	</div>
+</template>
+
+<script lang="ts">
+import { genericHelpers } from '@/mixins/genericHelpers';
+import type { IFormInputs } from '@/Interface';
+import Logo from '../components/Logo.vue';
+import {
+	MFA_AUTHENTICATION_RECOVERY_CODE_INPUT_MAX_LENGTH,
+	MFA_AUTHENTICATION_TOKEN_INPUT_MAX_LENGTH,
+	VIEWS,
+} from '@/constants';
+import { useUsersStore } from '@/stores/users.store';
+import { mapStores } from 'pinia';
+import { mfaEventBus } from '@/event-bus';
+import { defineComponent } from 'vue';
+import { useToast } from '@/composables/useToast';
+
+export default defineComponent({
+	name: 'MfaView',
+	mixins: [genericHelpers],
+	components: {
+		Logo,
+	},
+	async mounted() {
+		const { email, password } = window.history.state;
+		this.email = email;
+		this.password = password;
+		this.formInputs = [
+			{
+				name: 'token',
+				initialValue: '',
+				properties: {
+					label: this.$locale.baseText('mfa.code.input.label'),
+					placeholder: this.$locale.baseText('mfa.code.input.placeholder'),
+					maxlength: MFA_AUTHENTICATION_TOKEN_INPUT_MAX_LENGTH,
+					capitalize: true,
+					validateOnBlur: false,
+				},
+			},
+		];
+	},
+	setup() {
+		return {
+			...useToast(),
+		};
+	},
+	data() {
+		return {
+			email: '',
+			password: '',
+			hasAnyChanges: false,
+			formBus: mfaEventBus,
+			formInputs: null as null | IFormInputs,
+			showRecoveryCodeForm: false,
+			formError: '',
+			verifyingMfaToken: false,
+		};
+	},
+	computed: {
+		...mapStores(useUsersStore),
+	},
+	methods: {
+		onRecoveryCodeClick() {
+			this.formError = '';
+			this.showRecoveryCodeForm = true;
+			this.hasAnyChanges = false;
+			this.formInputs = [
+				{
+					name: 'recoveryCode',
+					initialValue: '',
+					properties: {
+						label: this.$locale.baseText('mfa.recovery.input.label'),
+						placeholder: this.$locale.baseText('mfa.recovery.input.placeholder'),
+						maxlength: MFA_AUTHENTICATION_RECOVERY_CODE_INPUT_MAX_LENGTH,
+						capitalize: true,
+						validateOnBlur: false,
+					},
+				},
+			];
+		},
+		onBackClick() {
+			if (!this.showRecoveryCodeForm) {
+				void this.$router.push({ name: VIEWS.SIGNIN });
+				return;
+			}
+
+			this.showRecoveryCodeForm = false;
+			this.hasAnyChanges = true;
+			this.formInputs = [
```

Comment:
duplicate of above.. maybe we can have this and the other state as variables..

Reply:
One is for recovery code and the other for the token. Both possibilities are handled, so it's not a duplication?

Reply:
there's a token one in 96 no? https://github.com/n8n-io/n8n/pull/4767/files/0856a88f6d8f7d4f74e8b8adec4f090bdac4ec59#diff-a2de3b83a56edca71cb9738f8b35fc24d9619660523f91fcb5fead0fa1b97126R96

Reply:
would do this for both 
```
const TOKEN_INPUT = {...};
const RECOVERY_INPUT = {...};
```